### ISSUE TITLE
[App]: some config keys don't show up in the workspace

### ISSUE THREAD

**Issue by albertfgu** [wandb/wandb#6222]
### Current Behavior

Here's an example run: https://wandb.ai/bmr-test/tau-1/runs/t8wvgguz/overview?workspace=user-albert

It has some config keys such as `encoder.0='linear'`. However, attempting to find the key `encoder.0` when filtering or grouping the workspace's runs does not work.

### Expected Behavior

All keys should show up.

### Environment

OS:

Browsers:

Version:

**Comment by thanos-wandb** [comment]
Hi @albertfgu thank you for reporting this issue. Could you please provide us with a code snippet how you've logged `encoder.0` to try and recreate this issue? also, is it possible to share the `debug.log` and `debug-internal.log` file from this specific run? These files are under your local folder` wandb/run-<date>_<time>-<run-id>/logs` in the same directory where you’re running your code.

**Comment by albertfgu** [comment]
The debug logs are long and I couldn't see anything that was much helpful, although the end of `debug.log` does seem to print the config. I don't want to print the entire config in a public issue here, but the last two lines have the form
```
2023-09-04 15:21:53,580 INFO    MainThread:3014801 [wandb_run.py:_config_callback():1282] config_cb None None {print entire config dictionary}
```
and it does correctly print the part of the config corresponding to this issue, i.e. `'encoder': ['linear', {'_name_': 'timestamp_embedding', 'table': True, 'keys': [0, 1, 2, 3, 4, 5]}]`.
Let me know if there are other specific lines in the debug logs that you want to see.

The config logging is done through Pytorch Lightning, so I suppose the issue could somehow be on that end. The minimal example looks like:
```python
import lightning.pytorch as pl
from lightning.pytorch.loggers import WandbLogger
from omegaconf import DictConfig, OmegaConf  # Our configs are passed as an OmegaConf, but should be possible to reproduce with vanilla dict

class MyLightningModule(pl.LightningModule):
    def __init__(self, config):
        super().__init__()
        self.save_hyperparameters(OmegaConf.to_container(config))

logger = CustomWandbLogger(
    settings=wandb.Settings(start_method="fork"),
    **config.wandb,
)

trainer = pl.Trainer(logger=logger)
model = SequenceLightningModule(config)
trainer.fit(model)
```
I have not seen any problem of this form though despite having many other keys of a similar nature to the `encoder.0` (i.e., nested dicts and lists within the config) which is causing this issue.

**Comment by thanos-wandb** [comment]
Hi @albertfgu thanks so much for detailed information. There could be an issue with nested dictionaries, does your config look like:

```
config_dict = {"encoder": {0: "linear", 
                           1: {"name": 'time', "keys": [1, 2, 3, 4]}}}
```

Because this seems to allow grouping/filtering for `encoder.0`.

**Comment by thanos-wandb** [comment]
Hi @albertfgu I wanted to follow up with you regarding your support request as we have not heard back from you. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by ad12** [comment]
@thanos-wandb thanks for follow up on this! I am also running into this issue.

My config would look like this:
```
encoder:
  - linear
  - name: time
     keys: [1, 2, 3, 4]
```

**Comment by thanos-wandb** [comment]
Thank you @ad12 for additional input. I can recreate this issue on my end when logging a nested config with that format. Would it work for you to log any json configs as artifact files?

**Comment by ad12** [comment]
I don't think so. i primarily use the config for grouping runs by metadata (great feature btw!). so having the nested configs show up for grouping purposes would be quite helpful

**Comment by thanos-wandb** [comment]
Understood, thank you, @ad12, for sharing the use case with us! Regrettably, this feature is not currently supported. I have logged it as a feature request with our engineering teams, and I will provide updates on any progress here.
